plugins/modules/systemd_info.py

```diff
@@ -195,11 +205,29 @@
   }
 '''
 
+import fnmatch
 from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.systemd import systemd_runner
+
+
+def get_version(runner):
+    with runner("version") as ctx:
+        rc, stdout, stderr = ctx.run()
+    return stdout.strip()
 
 
-def run_command(module, cmd):
-    rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+def list_units(runner, types_value):
+    context = "list_units no_pager typ types all plain no_legend"
+    with runner(context) as ctx:
+        rc, stdout, stderr = ctx.run(types=types_value)
+    return stdout.strip()
+
+
+def show_unit_properties(runner, prop_list, unit):
+    props_str = ",".join(prop_list)
+    context = "show p props dashdash unit"
+    with runner(context) as ctx:
+        rc, stdout, stderr = ctx.run(props=props_str, unit=unit)
```

Comment:
... here it would become:
```suggestion
    context = "show props dashdash unit"
    with runner(context) as ctx:
        rc, stdout, stderr = ctx.run(props=props_list, unit=unit)
```